test/Makefile

```diff
@@ -0,0 +1,64 @@
+LIBDIR=../lib
+GTEST_MODULE=$(LIBDIR)/googletest
+GTEST_DIR=$(LIBDIR)/googletest/googletest
+GTEST_SOURCEDIR=$(GTEST_DIR)/src
+GTEST_HEADERS=$(GTEST_DIR)/include/gtest/*.h \
+              $(GTEST_DIR)/include/gtest/internal/*.h
+GTEST_SRCS_=$(GTEST_SOURCEDIR)/*.cc $(GTEST_SOURCEDIR)/*.h $(GTEST_HEADERS)
+
+CPP_NETLIB_DIR=$(LIBDIR)/cpp-netlib
+YAML_CPP_DIR=$(LIBDIR)/yaml-cpp
+
+CPPFLAGS+=-isystem $(GTEST_DIR)/include
+CXXFLAGS=-std=c++11 -g -pthread
+
+# Where to find user code.
+SRC_DIR=../src
+SRCS=\
+     $(SRC_DIR)/format.cc
+OBJS=$(subst src/,test/,$(subst .cc,.o,$(SRCS)))
+
+TEST_DIR=.
+TESTS=$(wildcard $(TEST_DIR)/*.cc)
+TEST_OBJS=$(TESTS:%.cc=%.o)
+
+GTEST_LIBS=gtest_main.a
+
+all: build-googletest unittest
+
+clean:
+	rm -f unittest *.a *.o
+
+purge: clean
+	git submodule deinit -f --all
+
+init-submodules:
```

Comment:
Unless you want to re-init the submodules all the time, you shouldn't make `init-submodules` a phony target.

Reply:
You need to also `touch` it, like [this](https://github.com/supriyagarg/metadata-agent/blob/master/src/Makefile#L132).